model/ftn/w3iorsmd.ftn

```diff
@@ -995,23 +1005,12 @@
             IF (OARST) THEN
 !/DEBUGINIT         WRITE(740+IAPROC,*) 'Before reading CUR'
               IF ( FLOGRR(1,2) ) THEN
-                READ (NDSR,ERR=802,IOSTAT=IERR) TMP(1:NSEA)
-                READ (NDSR,ERR=802,IOSTAT=IERR) TMP2(1:NSEA)
-                DO I=1, NSEALM
-                  J = IAPROC + (I-1)*NAPROC
-                  IF (J .LE. NSEA) THEN
-                    CX(I) = TMP(J)
-                    CY(I) = TMP2(J)
-                  ENDIF
-                ENDDO
+                READ (NDSR,ERR=802,IOSTAT=IERR) CX(1:NSEA)
```

Comment:
Maybe I'm missing something obvious, but I don't understand why this was changed for CX,CY and ICEF but then you still leave the NSEALM sections for other parts below?

Reply:
CX, CY, and ICEF are input fields, and therefore each processor contains information on the field for the whole grid (index in the range 1-NSEA). This is why we do not need a XCX pointer to gather/scatter information between processors. In the case of the rest of the fields, each processor only contains information for part of the domain, with index in the range 1-NSEALM, and we need X pointers (such as XTAUOCX).

Reply:
If CX, CY are input fields, usually aren't those on (NX,NY) domains, not (1:NSEA)? I don't see an if SMC grid if statement, but maybe I haven't gone back far enough?

Reply:
Yes, current components is an input field and in principle they have dimensions (NX,NY), so when they are read they are stored in variables with those dimensions, in this case CX0, CY0, CXN, and CYN using the routine W3FLDG (which also does time interpolation if necessary):

https://github.com/ukmo-waves/WW3/blob/73ea7e11167863ef6f5e679351d1c9b929096054/model/ftn/ww3_shel.ftn#L2182-L2187

Once the input has been read, it is copied to each processor in the working variables CX and CY in routine W3UCUR, which have dimensions (0,NSEA):

https://github.com/ukmo-waves/WW3/blob/73ea7e11167863ef6f5e679351d1c9b929096054/model/ftn/w3adatmd.ftn#L922-L925

Reply:
So why is CX,CY being saved instead of CX0, CY0 and/or CXN, CYN?   Wouldn't CX, CY just get over written based on what CX0, CY0, CXN, and CYN?

Reply:
CX and CY are saved because they are the variables actually used by WWIII in all relevant calculations. CX and CY will only be overwritten by what is is CX0,CY0,CXN, and CYN if we force WWIII with currents. Even if this is the case, it is necessary to save the currents in the restart file in case we are sending via coupling at T+0 the currents to an atmospheric model, because the update of the currents only happens after the first coupling step. If the currents are not in the restart file, then the currents that will be sent to the atmosphere model will be zero.